src/main/java/io/jenkins/plugins/orka/client/OrkaClient.java

```diff
@@ -49,46 +45,36 @@ public OrkaClient(String endpoint, String email, String password) throws IOExcep
     public OrkaClient(String endpoint, String email, String password, int httpClientTimeout) throws IOException {
         this.client = clientBase.newBuilder().readTimeout(httpClientTimeout, TimeUnit.SECONDS).build();
         this.endpoint = endpoint;
-        this.token = this.getToken(email, password);
+        this.tokenResponse = this.getToken(email, password);
+
+        this.verifyToken();
     }
 
-    public List<VMResponse> getVMs() throws IOException {
-        String response = this.get(this.endpoint + VM_PATH + LIST_PATH);
+    public VMResponse getVMs() throws IOException {
```

Comment:
nit: We can name this `getVMsResponse` and this way will avoid `.getVMs().getVMs()`

Reply:
Yeah. I do not like it either.
I want to change it in another PR as this will change a lot of usages.